packages/block-library/src/footnotes/format.js

```diff
@@ -44,13 +48,26 @@ export const format = {
 		const registry = useRegistry();
 		const {
 			getSelectedBlockClientId,
+			getBlocks,
 			getBlockRootClientId,
 			getBlockName,
-			getBlocks,
+			getBlockParentsByBlockName,
 		} = useSelect( blockEditorStore );
 		const footnotesBlockType = useSelect( ( select ) =>
 			select( blocksStore ).getBlockType( name )
 		);
+		const hasParentCoreBlocks = useSelect( ( select ) => {
```

Comment:
Another tiny nit (not a blocker), would it be worth adding a comment before this `useSelect` to explain the decision to have a separate `useSelect`? I could imagine someone else in the future looking at this and thinking they could optimise it by re-using the `getSelectedBlockClientId` and `getBlockParentsByBlockName` functions above, and accidentally adversely affect performance in the process.